CoreFindView/src/au/gov/asd/tac/constellation/views/find2/components/AdvancedFindTab.java

```diff
@@ -97,25 +96,18 @@
     private final Button findPrevButton = new Button("Find Previous");
     private final Button findAllButton = new Button("Find All");
 
-    private static final long ATTRIBUTE_MODIFICATION_COUNTER = Long.MIN_VALUE;
-
     public AdvancedFindTab(final FindViewTabs parentComponent) {
         this.parentComponent = parentComponent;
         setText("Advanced Find");
         setGridContent();
         addCriteriaPaneButton.setOnAction(action -> addCriteriaPane(getSelectedGraphElementType()));
 
-        // Change the
-        lookForChoiceBox.getSelectionModel().selectedItemProperty().addListener((ObservableValue<? extends String> observableValue, final String oldElement, final String newElement) -> {
+        // Change the displayed list based on the graph element type selection
+        lookForChoiceBox.getSelectionModel().selectedItemProperty().addListener((final ObservableValue<? extends String> observableValue, final String oldElement, final String newElement) -> {
             changeDisplayedList(newElement);
         });
 
-        currentSelectionChoiceBox.getSelectionModel().selectedItemProperty().addListener(new ChangeListener<String>() {
-            @Override
-            public void changed(ObservableValue<? extends String> observableValue, String oldElement, String newElement) {
-                updateSelectionFactors();
-            }
-        });
+        currentSelectionChoiceBox.getSelectionModel().selectedItemProperty().addListener(action -> updateSelectionFactors());
```

Comment:
I think the lambda needs to have 3 parameters rather than just 1 to match what the anonymous function had. Else I think you're adding an invalidation listener rather than a change listener

Unless of course that was intentional in which case feel free to ignore this suggestion

Reply:
Hey @antares1470 are you able to clarify what you mean here? The intended action of the line is to just call updateSelectionFactors() when the selection changes.

Reply:
@Atlas139mkm so you're previous inner class function had 3 parameters: `observableValue`, `oldElement`, and `newElement`. The names aren't necessarily important here but point is there are 3. When you convert to a lambda function, the parameters of the lambda function should match those of that function as that is how it determines what is being added (in this case a `ChangeListener`). 

With your lambda function though, it only has one parameter. Consequently, rather than mapping to a `ChangeListener`, it is mapping to an `InvalidationListener` (whose function only has one parameter) which is the other kind of listener you can add here. Both kinds of listeners respond to different things and so consequently your code behaviour has changed, and by the sounds of it away from the intended behaviour.

Does that makes sense?